metrics/apis/metrics/v1alpha1/types.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// resource usage metrics of a node.
+type NodeMetrics struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty"`
+
+	// The following fields define time interval from which metrics were
+	// collected in the following format [Timestamp-Window, Timestamp].
+	Timestamp unversioned.Time     `json:"timestamp"`
+	Window    unversioned.Duration `json:"window"`
+
+	// The memory usage is the memory working set.
+	Usage v1.ResourceList `json:"usage"`
+}
+
+// resource usage metrics of a pod.
+type PodMetrics struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty"`
+
+	// The following fields define time interval from which metrics were
+	// collected in the following format [Timestamp-Window, Timestamp].
+	Timestamp unversioned.Time     `json:"timestamp"`
+	Window    unversioned.Duration `json:"window"`
+
+	// Metrics for all containers are collected within the same time window.
+	Containers []ContainerMetrics `json:"containers"`
+}
+
+// resource usage metrics of a container.
+type ContainerMetrics struct {
+	// Container name corresponding to the one from v1.Pod.Spec.Containers.
```

Comment:
Comments on fields shouldn't reference to Go types, since the comment is turned into swagger documentation and the user doesn't know the Go types. I guess you can say "corresponding to the one from pod.spec.containers" -- at least that is a little closer to the json key names (because it uses lower case and omits the v1)